### ISSUE TITLE
Bash launcher script ruins Cygwin terminal

### ISSUE THREAD

**Issue by GreyCat** [sbt/sbt-native-packager#1117]
Bash launcher script, generated out of a template, includes [these lines](https://github.com/sbt/sbt-native-packager/blob/c770faa17bb22032b3b165b0c5416851fc593052/src/main/resources/com/typesafe/sbt/packager/archetypes/scripts/bash-template#L225-L230). For a generic console application, if we're running it in Cygwin terminal, these lines effectively ruin the console after the run is finished by disabling echo. Here's a user reporting it for our app: https://github.com/kaitai-io/kaitai_struct/issues/360

There is a counterpart that's supposed to reset the terminal back [here](https://github.com/sbt/sbt-native-packager/blob/c770faa17bb22032b3b165b0c5416851fc593052/src/main/resources/com/typesafe/sbt/packager/archetypes/scripts/bash-template#L259), but it doesn't seem help (or may be it's even not  executed).

Removing that `stty ... -echo` line solves the issue. I don't know for certain why these lines were introduced, but I can make a guess that these were introduced as workaround for some jline-based application (which probably fixes terminal by itself on exit). For a normal application, this call just disables echo altogether: both during the application run (which ruins simple inputs like one with `scala.io.StdIn.readLine()`), and all apps (i.e. the shell) experience after the application has finished.

From my perspective, there are several ways we could solve it:

* The "right" way: remove these lines altogether. If that's something specific that jline needs, it's logical to do that workaround in jline, not in sbt launcher.
* The "complex" way: generate / run these lines only when jline is used in the app. I'm not sure that would be easy to detect, though.
* The "least invasive" way: Investigate why restoring echo does not help (or may be it's not getting executed at all?). Obviously, that's very unrealible anyway: i.e., simple signal which kills both the launcher script and the JVM, would still ruin the terminal.

## Expected behaviour

* During the run of a generated application in Cygwin, reading from Console works with echo.
* After running a generated application in Cygwin, console is in the same state it was before the run.

## Actual behaviour

* During the run of a generated application in Cygwin, reading from Console has disabled echo.
* After running a generated application in Cygwin, console is ruined and requires some reset (`reset`, `stty sane`, `stty echo`) to work interactively.

## Information

- What sbt-native-packager are you using: 1.3.5-SNAPSHOT
- What sbt version: 1.1.1
- What is your build system (e.g. Ubuntu, MacOS, Windows, Debian): Linux
- What package are you building (e.g. docker, rpm, ...): universal zip
- What version has your build tool (find out with e.g. `rpm --version`): N/A
- What is your target system (e.g. Ubuntu 16.04, CentOS 7): Windows 10, Cygwin terminal

**Comment by muuki88** [comment]
Thanks a lot for the detailed bug report along with some options to fix this.

> remove these lines altogether. If that's something specific that jline needs, it's logical to do that workaround in jline, not in sbt launcher

I agree. Unfortunately I have no clue if this was really the case why this line was added.
However if this breaks anything and jline wasn't able to provide a workaround in the last 5 years, we still can provide documentation for users that require it.

Are you able to test a fix? I haven't used windows in a long time. And actually never used cygwin.

**Comment by GreyCat** [comment]
Yes, I think I can test the fix. However, I'm not sure who might be using these jline hacks — i.e. if interactive sbt relies on it, or, for example, may be we should ask @lihaoyi of [Ammonite project](http://ammonite.io/), as Ammonite seems to [use jline](https://github.com/lihaoyi/Ammonite/blob/master/build.sbt#L263-L264) for its interactive shells?
